exist-core/src/main/java/org/exist/source/SourceFactory.java

```diff
@@ -111,7 +111,7 @@ public class SourceFactory {
                 && ((location.startsWith("/db") && !Files.exists(Paths.get(firstPathSegment(location))))
                 || (contextPath != null && contextPath.startsWith("/db") && !Files.exists(Paths.get(firstPathSegment(contextPath)))))) {
             final XmldbURI pathUri;
-            if (contextPath == null) {
+            if (contextPath == null || ".".equals(contextPath)) {
```

Comment:
I don't want to upset anyone, but I just wanted to point out that this is the wrong approach to fixing this issue.

Reply:
This is fixing an additional bug I found while creating tests:

see `importLibraryFromDbLocation` and `importLibraryFromXMLDBLocation`

Reply:
Yes, I understood that, but it still is the wrong approach.

Reply:
> Yes, I understood that, but it still is the wrong approach.

@adamretter Could you explain why you think this is the wrong approach and how do you think it should be solved correctly?

Reply:
`SourceFactory.getSource` is called with contextPath set to `""`, `"."` and `"/"` throughout the codebase in exist-db.
`"."` being the **default** value for `XQueryContext.moduleLoadPath`.
Its unit tests do not test for any of these (see https://github.com/eXist-db/exist/blob/develop/exist-core/src/test/java/org/exist/source/SourceFactoryTest.java).

Reply:
From all that I know the SourceFactory would like to get a `null` for contextPath but it isn't getting that except for a few call-sites where that is hard-coded.

Reply:
The null is significant

Reply:
I agree